### PR TITLE
Use gatsby-image for homepage's event list

### PR DESCRIPTION
**Pull request by TimothyGu**
<details><summary>outdated</summary>Is this something we would want to do? I thought the plan was to eventually use Facebook's events for this purpose, in which case this would not be appropriate. Does that remain the plan of record?</details>

Refs: #36

### CODE REVIEW

**Comment by GalenWong** [comment]
I have never thought of using facebook event for this. Maybe we discussed it before but I just can't remember. At least for me, I never plan to use facebook event for this.

If we were to use facebook events, how would we get the data? The facebook API is now quite restricted and I am not sure how easy it would be to get the data we want.

**Comment by TimothyGu** [comment]
The original thinking was Facebook's API, yes. That may indeed no longer be possible.

In that case, we should merge this PR.

**Comment by TimothyGu** [comment]
This has been rebased.

File: gatsby-node.js [hunk 1]
```diff
@@ -104,3 +106,30 @@ exports.onCreateNode = ({ node, actions, getNode }) => {
...
+
+exports.sourceNodes = ({ actions, createNodeId, createContentDigest }) => {
+	const promises = [];
+
+	const today = moment().hour(0).minute(0).second(0);
+	for (const rawEvent of events) {
+		promises.push(actions.createNode({
+			...rawEvent,
+			// eslint-disable-next-line camelcase
+			imgFile___NODE___relativePath: rawEvent.imgFilePath,
```
Comment (GalenWong) [inline gatsby-node.js]:
```suggestion
			// By specifying imgFile, we link a child of type File to this node with given relative path as imgFilePath.
			imgFile___NODE___relativePath: rawEvent.imgFilePath,
```

File: src/components/eventpage/eventpage.js [hunk 2]
```diff
@@ -48,34 +46,35 @@ const styles = theme => ({
...
-	const pastEvents = getPastEvents(eventsData);
-	// pastEvents are references to the same object
-	// therefore we can use indexOf.
-	const futureEvents = eventsData.filter(x =>
-		pastEvents.indexOf(x) === -1);
+	const data = useStaticQuery(graphql`
+		{
+			upcomingEvents: allHackEvent(sort: {fields: date}, filter: {past: {eq: false}}) {
+				nodes {
+					id
```
Comment (GalenWong) [inline src/components/eventpage/eventpage.js]:
```suggestion
```

File: src/data/events/events.js [hunk 3]
```diff
@@ -1,37 +1,34 @@
...
- */
 const events = [
 	{
 		name: 'JavaScript Chat with ACM Hack',
-		date: new Date(2019, 9, 8, 16, 0),
-		location: 'Engineering 6 Room 372',
-		imgURL: JSChat
+		date: getDateTime(2019, 10, 8, 16),
+		location: 'Engineering VI 372',
+		imgFilePath: 'event/2019f-js-chat.png'
```
Comment (GalenWong) [inline src/data/events/events.js]:
```suggestion
		imgFilePath: 'event/2019f-js-chat.png',
		detailLink: 'https://www.facebook.com/events/517868152322071/'
```

File: gatsby-node.js (hunk 1 above)
Comment (TimothyGu) [reply gatsby-node.js]: Rewrote this a bit.

**Pull request merged**